youtube-video-element.js

```diff
@@ -0,0 +1,542 @@
+  }
+
+  get readyState() {
+    return this.#readyState;
+  }
+
+  // If the getter from SuperVideoElement is overridden, it's required to define
+  // the setter again too unless it's a read only property! It's a JS thing.
+
+  get src() {
+    return this.getAttribute('src');
+  }
+
+  set src(val) {
+    if (this.src == val) return;
+    this.setAttribute('src', val);
+  }
+
+  /* onStateChange
+    -1 (unstarted)
+    0 (ended)
+    1 (playing)
+    2 (paused)
+    3 (buffering)
+    5 (video cued).
+  */
+
+  get paused() {
+    if (!this.isLoaded) return !this.autoplay;
+    return [-1, 0, 2, 5].includes(this.api?.getPlayerState?.());
+  }
+
+  get duration() {
+    return this.api?.getDuration?.() ?? NaN;
+  }
+
+  get autoplay() {
+    return this.hasAttribute('autoplay');
+  }
+
+  set autoplay(val) {
+    if (this.autoplay == val) return;
+    if (val) this.setAttribute('autoplay', '');
+    else this.removeAttribute('autoplay');
+  }
+
+  get buffered() {
+    if (!this.isLoaded) return createTimeRanges();
+    const progress =
+      this.api?.getVideoLoadedFraction() * this.api?.getDuration();
+    if (progress > 0) {
+      return createTimeRanges(0, progress);
+    }
+    return createTimeRanges();
+  }
+
+  get controls() {
+    return this.hasAttribute('controls');
+  }
+
+  set controls(val) {
+    if (this.controls == val) return;
+    if (val) this.setAttribute('controls', '');
+    else this.removeAttribute('controls');
+  }
+
+  get currentTime() {
+    return this.api?.getCurrentTime?.() ?? 0;
+  }
+
+  set currentTime(val) {
+    if (this.currentTime == val) return;
+    this.loadComplete.then(() => {
+      this.api?.seekTo(val, true);
+      if (this.paused) {
+        this.pause();
+      }
+    });
+  }
+
+  set defaultMuted(val) {
+    if (this.defaultMuted == val) return;
+    if (val) this.setAttribute('muted', '');
+    else this.removeAttribute('muted');
+  }
+
+  get defaultMuted() {
+    return this.hasAttribute('muted');
+  }
+
+  get loop() {
+    return this.hasAttribute('loop');
+  }
+
+  set loop(val) {
+    if (this.loop == val) return;
+    if (val) this.setAttribute('loop', '');
+    else this.removeAttribute('loop');
+  }
+
+  set muted(val) {
+    if (this.muted == val) return;
+    this.loadComplete.then(() => {
+      val ? this.api?.mute() : this.api?.unMute();
+    });
+  }
+
+  get muted() {
+    if (!this.isLoaded) return this.defaultMuted;
+    return this.api?.isMuted?.();
+  }
+
+  get playbackRate() {
+    return this.api?.getPlaybackRate?.() ?? 1;
+  }
+
+  set playbackRate(val) {
+    if (this.playbackRate == val) return;
+    this.loadComplete.then(() => {
+      this.api?.setPlaybackRate(val);
+    });
+  }
+
+  get playsInline() {
+    return this.hasAttribute('playsinline');
+  }
+
+  set playsInline(val) {
+    if (this.playsInline == val) return;
+    if (val) this.setAttribute('playsinline', '');
+    else this.removeAttribute('playsinline');
+  }
+
+  get poster() {
```

Comment:
do we need to support `poster`? I'd expect that either the youtube embed will be showing it or that users will use media-chrome's poster image element.